### PR TITLE
Full script tag integration

### PR DESCRIPTION
**Pull request by lilaconlee**
### Summary

Adds a full script tag integration which should include all some parameters from `dropin.create`. (Limited `dataset` browser support limits us from a full script tag integration). Integration will look like:
```html
 <form id="checkout-form" action="script-tag-result.html">
   <script src=SOURCE
    data-braintree-dropin-authorization=AUTHORIZATION
    data-payment-option-priority='["paypal","card", "paypalCredit"]'
    data-locale="de_DE"
    data-paypal.flow="checkout"
    data-paypal.amount="10.00"
    data-paypal.currency="USD"
    data-paypal-credit.flow="vault"
   ></script>
   <input id="pay-button" type="submit" value="Pay">
 </form>
```
JSDoc (also includes examples): 
screen shot 2017-06-29 at 10 10 33 am

It would be cool to update the app to have query parameter customization like the JS demo does and we could flesh out the integration tests.

### Checklist

- [x] Added a changelog entry
- [x] Ran unit tests

### CODE REVIEW

File: src/lib/create-from-script-tag.js [hunk 1]
```diff
@@ -1,11 +1,31 @@
 'use strict';
 
+var assign = require('./assign').assign;
 var find = require('./find-parent-form');
 var uuid = require('./uuid');
 var DropinError = require('./dropin-error');
 
+function addCompositeKeyValuePairToObject(obj, key, value) {
+  var decomposedKeys = key.split('.');
```
Comment (crookedneighbor) [inline src/lib/create-from-script-tag.js]: Can we do `var rootKey = decomposedKeys[0];`?

Looks like we use the first first element throughout this function.

File: test/app/promise.html [hunk 2]
```diff
@@ -143,6 +143,7 @@
       try {
         jsonParsedValue = JSON.parse(value);
         value = jsonParsedValue;
+
```
Comment (crookedneighbor) [inline test/app/promise.html]: extra space here?

File: src/views/payment-sheet-views/card-view.js [hunk 3]
```diff
@@ -451,4 +453,12 @@ function capitalize(string) {
   return string[0].toUpperCase() + string.substr(1);
 }
 
+function normalizeStyles(styles) {
+  Object.keys(styles).forEach(function (style) {
+    var newKey = camelCaseToSnakeCase(style);
```
Comment (crookedneighbor) [inline src/views/payment-sheet-views/card-view.js]: We should correct the name of this function.

**Comment by crookedneighbor** [comment]
Let's add analytics for whether or not someone is using a script tag integration or not.

File: src/lib/create-from-script-tag.js [hunk 4]
```diff
@@ -32,10 +66,23 @@ function createFromScriptTag(createFunction, scriptTag) {
...
+
+    if (value == null) {
+      return;
+    }
+
+    addCompositeKeyValuePairToObject(createOptions, compositeKey, value);
+  });
+
+  createFunction(createOptions).then(function (instance) {
+    analytics.sendEvent(instance._client, 'integration-type.script-tag');
```
Comment (crookedneighbor) [inline src/lib/create-from-script-tag.js]: Not super pleased with grabbing the private prop `_client` here, but this was the easiest way to include an analytic.

File: src/lib/kebab-case-to-camel-case.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+function kebabCaseToCamelCase(kebab) {
```
Comment (lilaconlee) [inline src/lib/kebab-case-to-camel-case.js]: Curious why we changed this from what we had before

File: src/index.js [hunk 6]
```diff
@@ -1,15 +1,38 @@
...
  *
- * The script tag integration will intercept the form submission and attempt to tokenize the credit card. If the tokenization is successful, it will insert the payment method nonce representing the credit card into a hidden input with the name `payment_method_nonce` and then submit your form.
+ * When your form is submitted, Drop-in will intercept the form submission and attempt to tokenize the payment method. If the tokenization is successful, it will insert the payment method nonce into a hidden input with the name `payment_method_nonce` and then submit your form. If the tokenization is unsuccessful, a relevant error will be shown in the UI.
+ *
+ * Specify creation options as data attributes in your script tag, as shown in the examples below. The following configuration properties may be set: `locale`, `payment-option-priority`, `paypal.amount`, `paypal.currency`, `paypal.flow`, `paypal-credit.amount`, `paypal-credit.currency`, `paypal-credit.flow`.
```
Comment (lilaconlee) [inline src/index.js]: I'd be tempted to have a table with the full data attribute and a description unless we think that's redundant with the `create` options.

File: src/lib/create-from-script-tag.js (hunk 4 above)
Comment (lilaconlee) [inline src/lib/create-from-script-tag.js]: Think it makes the most sense given the circumstances, but I feel that.

File: src/views/payment-sheet-views/card-view.js [hunk 7]
```diff
@@ -443,12 +445,20 @@ CardView.prototype._isCardTypeSupported = function (cardType) {
...
 
 function capitalize(string) {
   return string[0].toUpperCase() + string.substr(1);
 }
 
+function normalizeStyles(styles) {
+  Object.keys(styles).forEach(function (style) {
+    var newKey = kebabCaseToCamelCase(style);
+
+    styles[newKey] = styles[style];
```
Comment (lilaconlee) [inline src/views/payment-sheet-views/card-view.js]: Think I'd be fine one-lining this, even though I was the one who split it apart in the first place.

File: src/lib/kebab-case-to-camel-case.js (hunk 5 above)
Comment (crookedneighbor) [inline src/lib/kebab-case-to-camel-case.js]: We didn't have it before. The utility function included in `src/views/payment-sheet-views/card-view.js` goes the other way, it converts from camelcase to kebab case.

Unless you mean something else?

Comment (lilaconlee) [inline src/lib/kebab-case-to-camel-case.js]: Totally thought this was the one from the card view

**Comment by crookedneighbor** [comment]
Let's talk to support and docs about it. I could be persuaded.

**Comment by lilaconlee** [comment]
Closed with abf0c23

**Pull request closed without merging**